src/vs/vscode.d.ts

```diff
@@ -2972,6 +2972,37 @@ declare namespace vscode {
 	}
 
 	/**
+	 * An individual terminal instance within the integrated terminal.
+	 */
+	export interface Terminal {
+
```

Comment:
In the interest of keeping the API slim, I don't think it's worth keeping this there since the `name` never changes after passed into `createTerminal`.